### PR TITLE
Improve workspace details page

### PR DESCRIPTION
**Pull request by olexii4**
### What does this PR do?
Rework workspace details components:
- [breadcrumbs component](https://www.patternfly.org/v4/documentation/react/components/breadcrumb) and workspace status component(workspaceDetails header)
- component that handles tabs switching and errors indicating
- dedicated component for each tab content
- buttons to save, cancel changes
- add a warning  message when need to restart for apply changes 

Implement the overview tab.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/17899
https://github.com/eclipse/che/issues/16645

Screenshot from 2020-10-07 13-26-45
Screenshot from 2020-10-04 03-46-02
Screenshot from 2020-10-04 03-47-41
Screenshot from 2020-10-04 03-47-05
Screenshot from 2020-10-04 03-48-06
Screenshot from 2020-10-04 03-50-39

Signed-off-by: Oleksii Orel <oorel@redhat.com>

### CODE REVIEW

File: src/pages/WorkspaceDetails/OverviewTab/WorkspaceName/index.tsx [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+              <React.Fragment>
+                {name}
+                <Button data-testid="handle-edit-mode-toggle" variant="plain"
+                  onClick={() => this.handleEditModeToggle()}>
+                  <PencilAltIcon />
+                </Button>
+              </React.Fragment>
+            )}
+          </Text>
+        </TextContent>
```
Comment (akurinnoy) [inline src/pages/WorkspaceDetails/OverviewTab/WorkspaceName/index.tsx]:
```suggestion
        {(!isEditMode) && (
          <TextContent>
            <Text className={styles.name}>
                {name}
                <Button data-testid="handle-edit-mode-toggle" variant="plain"
                  onClick={() => this.handleEditModeToggle()}>
                  <PencilAltIcon />
                </Button>
            </Text>
          </TextContent>
        )}
```

Comment (olexii4) [reply src/pages/WorkspaceDetails/OverviewTab/WorkspaceName/index.tsx]: fixed

**Review by sleshchenko — APPROVED** [review]
Basic workflows are tested and seem to work fine
For corner cases where I found some bugs - I'll create distinct issues since much good stuff is implemented and we can't prevent appearing it on 7.20 =)
So, approving this PR as a great start point for Workspace Details life :champagne:

**Review by akurinnoy — APPROVED** [review]
Taking into account the amount of work's been done in this PR I'd like to merge it as soon as possible. And, in case of any bug appears it will be addressed it as a separate issue.
Good job, Oleksii!

**Pull request merged**